addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -5537,6 +5541,11 @@ registry.ImageTools = ImageHandlerOption.extend({
         if (params.optionsPossibleValues.resetCrop) {
             return this._isCropped();
         }
+        // Only the 'Crop' widget is visible when image is not supported for style options.
+        if (params.optionsPossibleValues.crop) {
+            const img = this._getImg();
+            return isImageSupportedForStyle(img) || this._isImageSupportedForProcessing(img);
+        }
```

Comment:
@xO-Tx I am not understanding this, are you sure you adapted correctly the forward-port? Difficult for me to know since you rebased at the same time as fixing the conflict...

Reply:
@qsm-odoo  Here is the explanation for this update:

A- On 15.0 (after [1]):

We check options visibility on `ImageHandlerOption` using this code:

```javascript
_isAllowedOnAllImages() {
    return isImageSupportedForStyle(this._getImg());
},
```

```javascript
async _renderCustomXML(uiFragment) {
    const img = this._getImg();
    if (this._isAllowedOnAllImages()) {
        return;
    }
    if (!this.originalSrc || !this._isImageSupportedForProcessing(img)) {
        [...uiFragment.childNodes].forEach(node => node.remove());
        return;
    }
    ...
},
```

This way and when image reuploaded & supported for processing, all options from `imageTools` are shown (including crop) for t-field images. That's why the code was added to hide other options in '_computeWidgetVisibility' and keep crop only.

```javascript
// Only the 'Crop' widget is visible when image is not supported for style options.
if (Object.keys(params.optionsPossibleValues).some(methodName => ['transform', 'selectStyle'].includes(methodName))) {
    return isImageSupportedForStyle(this._getImg());
}
```
B- Now on saas-15.1 / On this PR:

`_isAllowedOnAllImages` removed in [2] since we moved visibility check to `ImageHandlerOption > _computeWidgetVisibility` [3]:

```javascript
_computeWidgetVisibility(widgetName, params) {
    if (this._isImageProcessingWidget(widgetName, params)) {
        const img = this._getImg();
        return this._isImageSupportedForProcessing(img);
    }
    return isImageSupportedForStyle(this._getImg());
},
```
So we get the following results on new code on `ImageTools > _computeWidgetVisibility`:

```javascript
// Only the 'Crop' widget is visible when image is not supported for style options.
if (params.optionsPossibleValues.crop) {
    const img = this._getImg();
    return isImageSupportedForStyle(img) || this._isImageSupportedForProcessing(img);
}
```

1- For t-field images (first time): crop hidden since (`this._isImageSupportedForProcessing(img) = false & isImageSupportedForStyle(img) = false`).

2- For t-field image (image uploaded again & image supported by `applyModifications`) > crop visible (since `this._isImageSupportedForProcessing(img) = true`) and other style options are hidden because the `_super` call will use [3].

3- The crop always visible when `isImageSupportedForStyle = true`.

Which is the same behaviour as in 15.0.
What do you think ?

[1]: https://github.com/odoo/odoo/pull/82226/commits/12f35c254e1001fe5db0f0744cdc6445b9f22221
[2]: https://github.com/odoo/odoo/pull/80208

Reply:
It seems ok but the left-untouched `// Only the 'Crop' widget is visible when image is not supported for style options.` seems strange above the new code, all your explanation here could be code comment in the end.

Let's leave that for a future refactoring, I'll test and merge now thanks :+1: